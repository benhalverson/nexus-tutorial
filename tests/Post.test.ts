import { createTestContext } from './__helpers';
const ctx = createTestContext();

it('ensures that a draft can be created and published', async () => {
  // Create a new draft
  const draftResult = await ctx.client.request(`
    mutation {
      createDraft(title: "Nexus", body: "...") {
        id
        title
        body
        published
      }
    }
  `);
  expect(draftResult).toMatchInlineSnapshot();
  // publish the draft
  const publishResult = await ctx.client.request(
    `
    mutation publishDraft($draftId: Int!) {
      publish(draftId: $draftId) {
        id
        title
        body
        published
      }
    }
  `,
    { draftId: draftResult.createDraft.id }
  );
  // Snapshot the published draft and expect `published` to be true
  expect(publishResult).toMatchInlineSnapshot();
});
