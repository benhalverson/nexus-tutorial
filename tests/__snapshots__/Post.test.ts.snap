// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ensures that a draft can be created and published 1`] = `
Object {
  "createDraft": Object {
    "body": "...",
    "id": 33,
    "published": false,
    "title": "Nexus",
  },
}
`;

exports[`ensures that a draft can be created and published 2`] = `
Object {
  "publish": Object {
    "body": "...",
    "id": 33,
    "published": true,
    "title": "Nexus",
  },
}
`;

exports[`ensures that a draft can be created and published 3`] = `
Array [
  Object {
    "body": "...",
    "id": 1,
    "published": false,
    "title": "Nexus",
  },
  Object {
    "body": "...",
    "id": 2,
    "published": true,
    "title": "Nexus",
  },
  Object {
    "body": "...",
    "id": 3,
    "published": true,
    "title": "Nexus",
  },
  Object {
    "body": "...",
    "id": 4,
    "published": true,
    "title": "Nexus",
  },
  Object {
    "body": "...",
    "id": 5,
    "published": true,
    "title": "Nexus",
  },
  Object {
    "body": "...",
    "id": 6,
    "published": true,
    "title": "Nexus",
  },
  Object {
    "body": "...",
    "id": 7,
    "published": true,
    "title": "Nexus",
  },
  Object {
    "body": "...",
    "id": 8,
    "published": true,
    "title": "Nexus",
  },
  Object {
    "body": "...",
    "id": 9,
    "published": true,
    "title": "Nexus",
  },
  Object {
    "body": "...",
    "id": 10,
    "published": true,
    "title": "Nexus",
  },
  Object {
    "body": "...",
    "id": 11,
    "published": true,
    "title": "Nexus",
  },
  Object {
    "body": "...",
    "id": 12,
    "published": true,
    "title": "Nexus",
  },
  Object {
    "body": "...",
    "id": 13,
    "published": true,
    "title": "Nexus",
  },
  Object {
    "body": "...",
    "id": 14,
    "published": true,
    "title": "Nexus",
  },
  Object {
    "body": "...",
    "id": 15,
    "published": true,
    "title": "Nexus",
  },
  Object {
    "body": "...",
    "id": 16,
    "published": true,
    "title": "Nexus",
  },
  Object {
    "body": "...",
    "id": 17,
    "published": true,
    "title": "Nexus",
  },
  Object {
    "body": "...",
    "id": 18,
    "published": true,
    "title": "Nexus",
  },
  Object {
    "body": "...",
    "id": 19,
    "published": true,
    "title": "Nexus",
  },
  Object {
    "body": "...",
    "id": 20,
    "published": true,
    "title": "Nexus",
  },
  Object {
    "body": "...",
    "id": 21,
    "published": true,
    "title": "Nexus",
  },
  Object {
    "body": "...",
    "id": 22,
    "published": true,
    "title": "Nexus",
  },
  Object {
    "body": "...",
    "id": 23,
    "published": true,
    "title": "Nexus",
  },
  Object {
    "body": "...",
    "id": 24,
    "published": true,
    "title": "Nexus",
  },
  Object {
    "body": "...",
    "id": 25,
    "published": true,
    "title": "Nexus",
  },
  Object {
    "body": "...",
    "id": 26,
    "published": true,
    "title": "Nexus",
  },
  Object {
    "body": "...",
    "id": 27,
    "published": true,
    "title": "Nexus",
  },
  Object {
    "body": "...",
    "id": 28,
    "published": true,
    "title": "Nexus",
  },
  Object {
    "body": "...",
    "id": 29,
    "published": true,
    "title": "Nexus",
  },
  Object {
    "body": "...",
    "id": 30,
    "published": true,
    "title": "Nexus",
  },
  Object {
    "body": "...",
    "id": 31,
    "published": true,
    "title": "Nexus",
  },
  Object {
    "body": "...",
    "id": 32,
    "published": true,
    "title": "Nexus",
  },
  Object {
    "body": "...",
    "id": 33,
    "published": true,
    "title": "Nexus",
  },
]
`;
